docs/server/features/indexes/secondary.md

```diff
@@ -0,0 +1,103 @@
+- Secondary indexes are built in the background and may take some time to complete the initial indexing, especially for large databases.
+- Secondary indexes currently support only category and event type indexing. Future releases may introduce additional index types.
+- Read and subscribe operations on secondary indexes need to use the log position for tracking progress instead of event numbers, as indexes do not provide sequential event numbers.
+
+In terms of storage and performance, using secondary indexes can lead to:
+- Reduced storage usage compared to link events created by system projections by up to 50%.
+- Improved read performance for queries that filter by category or event type, as the need for link resolution is eliminated. Our tests indicate up to 10 times faster reads for such queries, but only with larger page sizes. We recommend reading at least 1000 records for regular read operations. Subscriptions to secondary indexes use 2048 as the default page size compared with 32 for regular subscriptions, which helps improve performance for subscription use cases.
+
+## Limitations
+
+The first version of secondary indexes has some limitations that will be addressed in future releases:
+- Deletion of events does not remove entries from secondary indexes. This may lead to index entries pointing to non-existent events. It might impact speed of read operations but won't produce any errors as index records that cannot be resolved would be skipped on the server side. Future releases will implement automatic cleanup of index entries when events are deleted.
+- The previous point also indicates that reading from secondary indexes would not grant retention policies such as `MaxAge` or `MaxCount`. Events that have been deleted may still be returned by index reads until the index entries are cleaned up in a future release.
+- Because read and subscribe operations using secondary indexes currently rely on client API for reading `$all`, those operations can only be performed by users that are part of the `$admins` group. Future releases will enable reading from `$all` and secondary indexes without requiring admin privileges.
+
+## Backup and restore
+
+KurrentDB uses an embedded DuckDB for storing index data. The DuckDB database files are stored in the same directory as the main KurrentDB database files. Unlike KurrentDB database files, DuckDB files are not append-only and can be modified in place.
+
+Because of that, using file-system based backup solutions (e.g., file copies) may lead to inconsistent backups of secondary indexes if the DuckDB files are modified during the backup process. Therefore, the recommended way to perform backup and restore for KurrentDB v25.1 and later is to use volume snapshots instead of file copies. Volume snapshots ensure that all files, including DuckDB files, are captured in a consistent state.
```

Comment:
```suggestion
Because of that, using file-system based backup solutions (e.g., file copies) may lead to inconsistent backups of secondary indexes if the DuckDB files are modified during the backup process. Therefore, the recommended way to perform backup and restore for KurrentDB v25.1 and later is to use volume snapshots instead of file copies. Volume snapshots ensure that all files, including DuckDB files, are captured in a consistent state. A file copy backup can still be performed if the KurrentDB node is stopped first.
```